src/lists.c

```diff
@@ -1348,17 +1348,19 @@ Int IsSSortListDefault (
         return 2L;
     }
 
-    /* a list must be homogeneous to be strictly sorted                    */
-    if ( ! IS_HOMOG_LIST(list) ) {
+    /* get the first element                                               */
+    elm1 = ELM0_LIST( list, 1 );
```

Comment:
I am slightly puzzled by this change; couldn't you simply replace `IS_HOMOG_LIST` by `IS_DENSE_LIST`?

Reply:
because IsDenseList is a category

Reply:
Also (sorry for multiple comments) it's not clear to me it's worth using IS_DENSE_LIST here, because if we don't already know it we'll have to scan the list twice, and we are already going to pass the list once anyway.